### PR TITLE
[terra-application-navigation] Update docs to use MDX syntax

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Updated terra-application-navigation docs to use MDX syntax.

### CODE REVIEW

File: packages/terra-application-navigation/src/terra-dev-site/doc/example/ExampleApplication.jsx [hunk 1]
```diff
@@ -141,6 +142,7 @@ const ExampleApplication = ({ onAction }) => {
   const [activeKey, setActiveKey] = useState(navigationItems[0].key);
 
   return (
+    <ExampleWrapper>
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/terra-dev-site/doc/example/ExampleApplication.jsx]: Does this have lint errors for indentation?

Comment (saket2403) [reply packages/terra-application-navigation/src/terra-dev-site/doc/example/ExampleApplication.jsx]: No

Comment (mjhenkes) [reply packages/terra-application-navigation/src/terra-dev-site/doc/example/ExampleApplication.jsx]: It should, Please fix the indentation.

Comment (saket2403) [reply packages/terra-application-navigation/src/terra-dev-site/doc/example/ExampleApplication.jsx]: Fixed 492f931

**Comment by avinashg1994** [comment]
An exception is being thrown when either: _**Settings, Help or logout**_ utility buttons are clicked from the utility dropdown in the [doc example](https://terra-framework-deploy-pr-1026.herokuapp.com/components/terra-application-navigation/application-navigation/example), also the "current action" is not being displayed.
This seems to work as expected on the [engineering site](https://engineering.cerner.com/terra-framework/components/terra-application-navigation/application-navigation/example)

**Comment by saket2403** [comment]
> An exception is being thrown when either: _**Settings, Help or logout**_ utility buttons are clicked from the utility dropdown in the [doc example](https://terra-framework-deploy-pr-1026.herokuapp.com/components/terra-application-navigation/application-navigation/example), also the "current action" is not being displayed.
> This seems to work as expected on the [engineering site](https://engineering.cerner.com/terra-framework/components/terra-application-navigation/application-navigation/example)

Fixed b967176

File: packages/terra-application-navigation/src/terra-dev-site/doc/application-navigation/About.1.doc.mdx [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+
+* [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
+* [Responsive Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#responsive-support)
+* [Mobile Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#mobile-support)
+* [Internationalization Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#internationalization-i18n-support)
+* [Localization Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#internationalization-i18n-support)
+
+[View component source code](https://github.com/cerner/terra-framework/tree/master/packages/terra-application-navigation)
+
 ## Prerequisites
```
Comment (avinashg1994) [inline packages/terra-application-navigation/src/terra-dev-site/doc/application-navigation/About.1.doc.mdx]: This can be renamed as the "Usage" section

**Pull request merged**